apache-kafka-2/src/test/java/com/baeldung/kafka/message/ordering/SinglePartitionIntegrationTest.java

```diff
@@ -0,0 +1,121 @@
+        consumerProperties.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, JacksonDeserializer.class.getName());
+        consumerProperties.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+        consumerProperties.put(Config.CONSUMER_VALUE_DESERIALIZER_SERIALIZED_CLASS, UserEvent.class);
+        consumerProperties.put(ConsumerConfig.GROUP_ID_CONFIG, "test-group");
+        consumer = new KafkaConsumer<>(consumerProperties);
+        admin = Admin.create(adminProperties);
+
+
+        List<NewTopic> topicList = new ArrayList<>();
+        NewTopic newTopic = new NewTopic(Config.SINGLE_PARTITION_TOPIC, Config.SINGLE_PARTITION, Config.REPLICATION_FACTOR);
+        topicList.add(newTopic);
+        CreateTopicsResult result = admin.createTopics(topicList);
+        KafkaFuture<Void> future = result.values().get(Config.SINGLE_PARTITION_TOPIC);
+        future.whenComplete((voidResult, exception) -> {
+            if (exception != null) {
+                System.err.println("Error creating the topic: " + exception.getMessage());
+            } else {
+                System.out.println("Topic created successfully!");
+            }
+        }).get();
+    }
+
+    @AfterAll
+    static void destroy() {
+        KAFKA_CONTAINER.stop();
+    }
+
+    @Test
+    void givenASinglePartition_whenPublishedToKafkaAndConsumed_thenCheckForMessageOrder() throws ExecutionException, InterruptedException {
+        List<UserEvent> sentUserEventList = new ArrayList<>();
+        List<UserEvent> receivedUserEventList = new ArrayList<>();
+        for (long count = 1; count <= 10; count++) {
+            UserEvent userEvent = new UserEvent(UUID.randomUUID().toString());
+            userEvent.setEventNanoTime(System.nanoTime());
+            ProducerRecord<Long, UserEvent> producerRecord = new ProducerRecord<>(Config.SINGLE_PARTITION_TOPIC, userEvent);
+            Future<RecordMetadata> future = producer.send(producerRecord);
+            sentUserEventList.add(userEvent);
+            RecordMetadata metadata = future.get();
+            System.out.println("User Event ID: " + userEvent.getUserEventId() + ", Partition : " + metadata.partition());
+        }
+
+        consumer.subscribe(Collections.singletonList(Config.SINGLE_PARTITION_TOPIC));
+        ConsumerRecords<Long, UserEvent> records = consumer.poll(TIMEOUT_WAIT_FOR_MESSAGES);
+        records.forEach(record -> {
+            UserEvent userEvent = record.value();
+            receivedUserEventList.add(userEvent);
+            System.out.println("User Event ID: " + userEvent.getUserEventId());
+        });
+        boolean result = true;
+        for (int count = 0; count <= 9; count++) {
+            UserEvent sentUserEvent = sentUserEventList.get(count);
+            UserEvent receivedUserEvent = receivedUserEventList.get(count);
+            if (!sentUserEvent.equals(receivedUserEvent) && result) {
+                result = false;
+            }
+        }
+        assertTrue(result);
```

Comment:
For complex assertions, we use AssertJ, for example:
```suggestion
        assertThat(receivedUserEventList)
            .isEqualTo(sentUserEventList)
            .containsExactlyElementsOf(sentUserEventList);
```